operator/merkle_tree/lib/src/lib.rs

```diff
@@ -1,21 +1,35 @@
 use lambdaworks_crypto::merkle_tree::merkle::MerkleTree;
-use batcher::types::{VerificationBatch, VerificationData};
+use aligned_sdk::types::{VerificationData, VerificationDataCommitment, VerificationCommitmentBatch};
 
-// TODO: Ten times the size of one proof, could be changed later
-const MAX_BATCH_SIZE: usize = 2 * 1024 * 1024 * 10;
 
 #[no_mangle]
 pub extern "C" fn verify_merkle_tree_batch_ffi(
-    batch_bytes: &[u8; MAX_BATCH_SIZE],
+    batch_ptr: *const u8,
     batch_len: usize,
     merkle_root: &[u8; 32]
 ) -> bool {
-
-    let batch: Vec<VerificationData> = serde_json::from_slice(&batch_bytes[..batch_len]).unwrap();
-    let batch_merkle_tree: MerkleTree<VerificationBatch> = MerkleTree::build(&batch);
-    let batch_merkle_root = hex::encode(batch_merkle_tree.root);
+    if batch_ptr.is_null() || batch_len == 0 {
+        return false;
+    }
+    
+    let batch_bytes = unsafe { std::slice::from_raw_parts(batch_ptr, batch_len) };
+
+    let batch = match serde_json::from_slice::<Vec<VerificationData>>(batch_bytes) {
+        Ok(batch) => batch,
+        Err(_e) => {
+            return false;
+        }
+    };
+
+    let batch_data_comm: Vec<VerificationDataCommitment> = batch.into_iter()
+                                                            .map(VerificationDataCommitment::from)
+                                                            .collect();
+
+    let computed_batch_merkle_tree: MerkleTree<VerificationCommitmentBatch> = MerkleTree::build(&batch_data_comm);
+    let computed_batch_merkle_root = hex::encode(computed_batch_merkle_tree.root);
     let received_merkle_root = hex::encode(merkle_root);
-    return batch_merkle_root == received_merkle_root;
+
+    return computed_batch_merkle_root == received_merkle_root;
```

Comment:
I think you can check this without unnecessarily encoding to hex

Reply:
It doesn't compile without `hex::encode()`

Reply:
```rust
let computed_batch_merkle_tree: MerkleTree<VerificationCommitmentBatch> = MerkleTree::build(&batch_data_comm);

return computed_batch_merkle_tree.root == *merkle_root;
```